src/cmd/pkg/auth/auth.go

```diff
@@ -0,0 +1,234 @@
+func NewCommand() *cobra.Command {
+	root := &cobra.Command{
+		Use:   "auth",
+		Short: "All commands related to authenticating the chronoctl session.",
+	}
+
+	root.AddGroup(groups.Commands)
+	c := subcommand{}
+	root.AddCommand(
+		c.newAuthLoginCmd(browser.OpenURL),
+		c.newSetDefaultOrgCmd(),
+		c.newPrintAccessTokenCmd(),
+		c.newListCmd(),
+	)
+	return root
+}
+
+type subcommand struct {
+	store *token.Store
+}
+
+// getStore returns a token store contained in the subcommand if one is defined, otherwise returns a NewChronoctlStore
+func (c *subcommand) getStore() (*token.Store, error) {
+	if c.store != nil {
+		return c.store, nil
+	}
+	store, err := NewChronoctlStore()
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	return store, nil
+}
+
+type loginOpts struct {
+	orgName           string
+	loginPath         string
+	skipSetDefaultOrg bool
+}
+
+func (o *loginOpts) registerFlags(cmd *cobra.Command) {
+	const loginPathFlag = "login-path"
+	cmd.Flags().StringVar(&o.orgName, "org-name", "", "The name of your team's Chronosphere organization. Defaults to "+env.ChronosphereOrgNameKey+" environment variable.")
+	cmd.Flags().StringVar(&o.loginPath, loginPathFlag, defaultLoginPath, "the path to the org's chronoctl login page")
+	cmd.Flags().MarkHidden(loginPathFlag) //nolint:errcheck
+	cmd.Flags().BoolVar(&o.skipSetDefaultOrg, "skip-set-default-org", false, "When true, will not the current org as default for future commands")
+}
+
+// browserOpenFunc is used to set the function that will be run to open the browser. Useful for testing with a fake browser.
+type browserOpenFunc func(URL string) error
+
+func (c *subcommand) newAuthLoginCmd(openFunc browserOpenFunc) *cobra.Command {
+	opts := &loginOpts{}
+	cmd := &cobra.Command{
+		Use:     "login",
+		GroupID: groups.Commands.ID,
+		Short:   "Authenticate the chronoctl session as your Chronosphere user.",
+		RunE: func(command *cobra.Command, strings []string) error {
+			store, err := c.getStore()
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			ls, err := newLoginServer(store, openFunc, opts)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			return ls.login(command.Context(), command.InOrStdin(), command.OutOrStdout(), command.ErrOrStderr())
+		},
+	}
+
+	opts.registerFlags(cmd)
+	return cmd
+}
+
+func (c *subcommand) newSetDefaultOrgCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "set-default-org ORG_NAME",
+		GroupID: groups.Commands.ID,
+		Short:   "Sets the default Chronosphere organization for future commands.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			store, err := c.getStore()
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			if len(args) != 1 {
+				return errors.WithStack(errMustIncludeOrgName)
+			}
+			return setDefaultOrg(store, args[0])
+		},
+	}
+	return cmd
+}
+
+func (c *subcommand) newPrintAccessTokenCmd() *cobra.Command {
+	cmd := &cobra.Command{
```

Comment:
Consider using https://pkg.go.dev/github.com/spf13/cobra#ExactArgs for arg length validation as it'll print out the usage.

Reply:
Thanks for the pointer! [`1d1a4cd` (#27)](https://github.com/chronosphereio/chronoctl-core/pull/27/commits/1d1a4cd2f4a7650d59c87dcc4336a2064336deff)